### DISCUSSION TITLE
蓝盾模板, 定时触发 希望支持自定义时间 或者随机时间

### DISCUSSION THREAD

**Discussion by 727155455** [Tencent/bk-ci#4328] [Ideas]
### 作为
DevOps 团队里的Dev，要增加 项目每天的定期部署
当前通过模板 已实例多个项目

### 我希望
模板里 定时触发 插件 支持 时间 通过变量定义, 或者 增加 随机时间

### 以便于
- 多个实例 分散部署, 避免同一时间并发部署

### 期望的验收标准
- 模板定义时间,各实例依据这个时间 上下自动浮动几分钟,避免并发部署 (最期望方案)
- 定时触发 时间可通过变量自定义 (可接受方案)

**Comment by zhenlinyang** [comment]
建议bk-ci可以把定时构建做到**基础设置**里。

**Comment by zanyzhao** [comment]
@zhenlinyang 我们有个规划，可以关注下 #1130

**Comment by fayewong-v5** [comment]
优化定时触发插件，支持用户输入变量 @carlyin0801 @zhanxu33

**Comment by cly0115** [comment]
希望支持通过变量设置时间

**Comment by zanyzhao** [comment]
再加上这个就更爽了 #2891

**Comment by robling** [comment]
一年了，真的很需要这个功能

**Comment by sandyskies** [comment]
我们也需要这个功能，在某个时间段，随机执行

**Comment by zanyzhao** [comment]
#4073 我们优先支持自定义时间，后续再评估“随机时间”的可行性
